chest/core.py

```diff
@@ -134,8 +174,25 @@ def __init__(self, data=None, path=None, available_memory=None,
         self._on_miss = on_miss
         self._on_overflow = on_overflow
 
-    def __str__(self):
-        return '<chest at %s>' % self.path
+    def __repr__(self):
+        return '<chest at %s: keys=%s>' % (self.path, list(self._keys))
+    __str__ = __repr__
+
+    @property
+    def disk_usage(self):
+        path = self.path
+        return sum(
+            os.path.getsize(os.path.join(r, f))
+            for r, _, fs in os.walk(path) for f in fs
+        )
+
+    def dump(self, data, file_):
+        tmp = BytesIO()
+        self._dump(data, tmp)
+        bs = tmp.getvalue()
+        while self.disk_usage + len(bs) > self.available_disk:
+            self.on_full(self)
```

Comment:
Perhaps we could maintain a total, like how we do for `memory_usage` and add or subtract from it as we add or remove files from disk.  

Alternatively can you measure the overhead of how this works when we have one million files?  I expect this to be non-negligible at that scale.